src/chunk.c

```diff
@@ -1185,7 +1174,10 @@ chunk_create_from_hypercube_after_lock(const Hypertable *ht, Hypercube *cube,
 			ts_internal_to_time_int64(cube->slices[0]->fd.range_start, dim->fd.column_type);
 		int64 range_end =
 			ts_internal_to_time_int64(cube->slices[0]->fd.range_end, dim->fd.column_type);
-		int chunk_exists = insert_func_ptr(ht->main_table_relid, range_start, range_end);
+
+		int chunk_exists =
+			callbacks->chunk_insert_check_hook(ht->main_table_relid, range_start, range_end);
+
 		if (chunk_exists)
```

Comment:
this block became conditional based on the presence of the state of `callbacks` above; and based on the result of the `hook` some check is being executed...I think the below block either belongs to the `new hook` ; or the old code should remain here....in its current form I feel its a bit odd..